ubuntu/ubuntu-20.04.pkr.hcl

```diff
@@ -0,0 +1,77 @@
+packer {
+  required_plugins {
+    ansible = {
+      source  = "github.com/hashicorp/ansible"
+      version = "~> 1"
+    }
+    qemu = {
+      source  = "github.com/hashicorp/qemu"
+      version = "~> 1"
+    }
+  }
+}
+
+variable "SSH_PRIVATE_KEY_FILE" {
+  type    = string
+  default = ""
+}
+
+variable "SSH_PUB_KEY" {
+  type    = string
+  default = ""
+}
+
+source "qemu" "ubuntu-20-04" {
+  boot_command              = [
+    "<enter><enter><f6><esc><wait>",
+    "<bs><bs><bs><bs><bs><bs><bs><bs>",
+    "<bs><bs><bs><bs><bs><bs><bs><bs>",
+    "<bs><bs><bs><bs><bs><bs><bs><bs>",
+    "<bs><bs><bs><bs><bs><bs><bs><bs>",
+    "<bs><bs><bs><bs><bs><bs><bs><bs>",
+    "/casper/vmlinuz ",
+    "initrd=/casper/initrd ",
+    " autoinstall ds=nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ ",
+    " PACKER_USER=ubuntu PACKER_AUTHORIZED_KEY={{ .SSHPublicKey | urlquery }}",
```

Comment:
This doesn't look right. You're setting an environment variable here, but this is the boot command. the user should be set outside of this. Let me have a look at the docs and see how to do it.

Reply:
Note that these environment variables are later used for SSH auth with `cloud-init`:

https://github.com/EGI-Federation/fedcloud-vmi-templates/blob/7f6fe98eef5140922ba635e7222ccf23a6eda7b6/ubuntu/httpdir/user-data#L28-L39

Reply:
This is also what the [docs](https://developer.hashicorp.com/packer/integrations/hashicorp/qemu/latest/components/builder/qemu#ssh-key-pair-automation) say:
![image](https://github.com/user-attachments/assets/0c57824d-0f8f-4535-92d4-7074696a328d)

Reply:
I am using Packer v1.11.2

Reply:
ah ok - am I correct that you're providing a custom ssh cert/key for the communicator?

Reply:
> Note that these environment variables are later used for SSH auth with `cloud-init`:
> 
> https://github.com/EGI-Federation/fedcloud-vmi-templates/blob/7f6fe98eef5140922ba635e7222ccf23a6eda7b6/ubuntu/httpdir/user-data#L28-L39

If you're using ansible to provision the machine, why do you need user-data?

Reply:
Yes we are using a custom ssh cert/key to avoid setting a password to any user (and forget later to remove it or leave any traces of it)
Ansible cannot ssh into the machine if there is no configured user, and users are populated via the user-data.

Reply:
Below is my understanding of this config, but @enolfc please jump and correct me!

> ah ok - am I correct that you're providing a custom ssh cert/key for the communicator?

Packer will generate a temporary SSH keypair for every build. What we do here is to grab it and inject it into the VM to connect with Ansible later on.

> If you're using ansible to provision the machine, why do you need user-data?

IMHO the main goal of `user-data` is used to inject the correct SSH key to connect with Ansible later. There is also the option to format the root disk but I believe we are using the default. The rest of the config currently done by `user-data` may indeed be moved to Ansible.

Reply:
fair enough! Let's see if we can't get that into the various files as comments, for future eyes.

Reply:
looks like I didn't see Enol's comment before replying, sorry!